source/MaterialXRuntime/Private/PvtStage.cpp

```diff
@@ -88,14 +88,63 @@ void PvtStage::removePrim(const PvtPath& path)
         throw ExceptionRuntimeError("Given path '" + path.asString() + " does not point to a prim in this stage");
     }
 
+    // Destroy this prim and all its children.
+    prim->destroy();
+
+    // Remove from its parent.
+    PvtPrim* parent = prim->getParent();
+    parent->removeChildPrim(prim);
+}
+
+void PvtStage::disposePrim(const PvtPath& path)
+{
+    PvtPrim* prim = getPrimAtPathLocal(path);
+    if (!(prim && prim->getParent()))
+    {
+        throw ExceptionRuntimeError("Given path '" + path.asString() + " does not point to a prim in this stage");
+    }
+
+    // Make sure the prim has no connections.
+    for (RtAttribute attr : prim->getAttributes())
+    {
+        if (attr.isA<RtInput>() && attr.asA<RtInput>().isConnected())
+        {
+            throw ExceptionRuntimeError("Found a connection to '" + attr.getName().str() + "'. Cannot dispose a prim with connections.");
+        }
+        else if (attr.isA<RtOutput>() && attr.asA<RtOutput>().isConnected())
+        {
+            throw ExceptionRuntimeError("Found a connection from '" + attr.getName().str() + "'. Cannot dispose a prim with connections.");
+        }
+    }
+
     // Dispose this prim and all its children.
-    prim->dispose();
+    prim->dispose(true);
 
     // Remove from its parent.
     PvtPrim* parent = prim->getParent();
     parent->removeChildPrim(prim);
 }
 
+void PvtStage::revivePrim(const PvtPath& parentPath, const RtPrim& primH)
+{
+    // Use explicit cast since the asA() method will throw
```

Comment:
Won't this cause a lot of overhead if a lot of prims are deleted.

Reply:
Which part do you mean would cause overhead?